### PR TITLE
Provide an API to register pre loaded npm packages

### LINKED ISSUE #1069: Provide an API to register pre loaded npm packages
steal.getNpmPackages / steal.addNpmPackages

### PR DESCRIPTION
**Pull request by m-mujica**
Closes #1069 

@matthewp is this what you had in mind?

### CODE REVIEW

**Comment by m-mujica** [comment]
and the change in steal-tools would be https://github.com/stealjs/steal-tools/pull/612/commits/c330ba14cd85d034e69ec967ce7781450a631ddc

File: ext/npm-extension.js [hunk 1]
```diff
@@ -389,6 +389,25 @@ exports.addExtension = function(System){
 		oldConfig.apply(loader, arguments);
 	};
 
+	if (typeof steal !== undefined) {
```
Comment (matthewp) [inline ext/npm-extension.js]: We sure import steal like `var steal = require("@steal");`

File: ext/npm-extension.js [hunk 2]
```diff
@@ -389,6 +389,25 @@ exports.addExtension = function(System){
...
+		steal.addNpmPackages = function(packages) {
+			packages = packages || [];
+
+			for (var pkg of packages) {
+				var path = pkg && pkg.fileUrl;
+
+				if (path) {
+					System.npmContext.paths[path] = pkg;
```
Comment (matthewp) [inline ext/npm-extension.js]: `this.loader` instead of `System`

File: ext/npm-extension.js [hunk 3]
```diff
@@ -389,6 +389,25 @@ exports.addExtension = function(System){
...
+				}
+			}
+		};
+
+		steal.getNpmPackages = function() {
+			var context = System.npmContext;
```
Comment (matthewp) [inline ext/npm-extension.js]: `this.loader.npmContext`

File: ext/npm-extension.js [hunk 4]
```diff
@@ -389,6 +389,25 @@ exports.addExtension = function(System){
 		oldConfig.apply(loader, arguments);
 	};
 
+	if (typeof steal !== undefined) {
+		steal.addNpmPackages = function(packages) {
+			packages = packages || [];
+
+			for (var pkg of packages) {
```
Comment (matthewp) [inline ext/npm-extension.js]: Use Object.keys().forEach or for/in, can't use for/of

Comment (m-mujica) [inline ext/npm-extension.js]: working on it

Comment (m-mujica) [inline ext/npm-extension.js]: working on it.

File: ext/npm-extension.js [hunk 5]
```diff
@@ -389,6 +390,25 @@ exports.addExtension = function(System){
...
+
+	steal.addNpmPackages = function(packages) {
+		packages = packages || [];
+
+		Object.keys(packages).forEach(function(key) {
+			var pkg = packages[key];
+			var path = pkg && pkg.fileUrl;
+
+			if (path) {
+				this.loader.npmContext.paths[path] = pkg;
```
Comment (matthewp) [inline ext/npm-extension.js]: wrong `this`

**Comment by m-mujica** [comment]
should be good to go now.

**Comment by m-mujica** [comment]
👍  or 👎

**Comment by matthewp** [comment]
LGTM 👍

**Pull request merged**